Source/JavaScriptCore/wasm/WasmOperationsInlines.h

```diff
@@ -192,6 +195,59 @@ inline void structSet(Instance* instance, EncodedJSValue encodedStructReference,
     return structPointer->set(globalObject, fieldIndex, toJSValue(globalObject, fieldType, argument));
 }
 
+inline bool refCast(Instance* instance, EncodedJSValue encodedReference, bool allowNull, int32_t heapType)
+{
+    JSValue refValue = JSValue::decode(encodedReference);
+    if (refValue.isNull())
+        return allowNull;
+
+    switch (static_cast<Wasm::TypeKind>(heapType)) {
+    case Wasm::TypeKind::Funcref:
+    case Wasm::TypeKind::Externref:
+        // Casts to funcref/externref canot fail as they are the top types of their respective hierarchies, and static type-checking does not allow cross-hierarchy casts.
+        return true;
+    case Wasm::TypeKind::I31ref:
+        return refValue.isInt32();
+    case Wasm::TypeKind::Arrayref:
+        return jsDynamicCast<JSWebAssemblyArray*>(refValue);
+    case Wasm::TypeKind::Structref:
+        return jsDynamicCast<JSWebAssemblyStruct*>(refValue);
+    default: {
+        ASSERT(!Wasm::typeIndexIsType(static_cast<Wasm::TypeIndex>(heapType)));
+        Wasm::TypeDefinition& signature = instance->module().moduleInformation().typeSignatures[heapType];
+        auto signatureRTT = instance->module().moduleInformation().rtts[heapType];
```

Comment:
We just always use the RTT from the module information now, as we don't store a list of Structures in the Wasm Module now.